### PR TITLE
Update logExporter to use rotating logs

### LINKED ISSUE #59: Android - Support Ditto automatic rotating logs in the Log Exporter tool
Today, the Log Exporter tool requires that users turn on Ditto logs to a local file, which they then have to manage to it does not grow unbounded.

Since this tool was initially created, we created an automatic log rotation system which is on by default - the log exporter should ideally use this system instead and not require users to manage an additional log file in app code

### PR DESCRIPTION
**Pull request by zmarkan**
Fixes https://github.com/getditto/DittoAndroidTools/issues/59 

now ready for review, and hopefully publishing

### CODE REVIEW

File: DittoExportLogs/build.gradle [hunk 1]
```diff
@@ -18,6 +18,7 @@ dependencies {
     implementation libs.androidx.compose.material3.material3
     implementation libs.androidx.compose.ui.ui
     implementation libs.androidx.navigation.navigationCompose
+    implementation(libs.live.ditto.ditto)
```
Comment (rajramsaroop) [inline DittoExportLogs/build.gradle:21]: You'll want to merge the latest main as it resolves some of these dependency issues, specifically with the ditto sdk

File: DittoExportLogs/src/main/java/live/ditto/dittoexportlogs/ExportLogs.kt [hunk 2]
```diff
@@ -1,26 +1,116 @@
...
+    val filePath = File(downloadsDir, fileName).absolutePath
+
+    AlertDialog(
+        modifier = modifier,
+        onDismissRequest = onDismiss,
+        title = {
+            Text(title)
+        },
+        text = {
+            Text("Logs will be exported to: \nDownloads/$fileName\"")
```
Comment (rajramsaroop) [inline DittoExportLogs/src/main/java/live/ditto/dittoexportlogs/ExportLogs.kt:61]: Let's stick this bad boy in a strings.xml

File: DittoExportLogs/src/main/java/live/ditto/dittoexportlogs/ExportLogs.kt [hunk 3]
```diff
@@ -1,26 +1,116 @@
...
+                            Toast.makeText(context, toastText, Toast.LENGTH_LONG).show()
+                        }
+                        catch (e: DittoError.IoError) {
+                            Log.e("DittoTools LogExporter", e.reason.toString())
+                            Toast.makeText(context, "Error exporting logs: ${e.message}", Toast.LENGTH_LONG).show()
+                        }
+                        finally {
+                            isLoading = false
+                            onDismiss()
+                        }
```
Comment (rajramsaroop) [inline DittoExportLogs/src/main/java/live/ditto/dittoexportlogs/ExportLogs.kt:85]: Not sure what sort of time crunch we are on here, but I'd prefer this to be in a view model

File: DittoToolsViewer/build.gradle.kts [hunk 4]
```diff
@@ -34,7 +34,8 @@ dependencies {
 
     implementation(libs.live.ditto.ditto)
     implementation(libs.live.ditto.databrowser)
-    implementation(libs.live.ditto.exportlogs)
+//    implementation(libs.live.ditto.exportlogs)
+    implementation(project(":DittoExportLogs"))
```
Comment (rajramsaroop) [inline DittoToolsViewer/build.gradle.kts:38]: 1. Don't forget to remove this before merging
2. This creates another issue that I think I've added to the tools backlog (to be figured out another time?), but if we uncomment out the code, it won't be pointing to the latest version of the tool - https://github.com/getditto/DittoAndroidTools/issues/123

File: DittoExportLogs/src/main/java/live/ditto/dittoexportlogs/ExportLogs.kt [hunk 5]
```diff
@@ -1,26 +1,116 @@
...
+                onClick = {
+
+
+                    coroutineScope.launch {
+                        isLoading = true
+                        try {
+                            val sizeWritten = DittoLogger.exportToFile(filePath)
+
+                            val sizeWrittenInKB = sizeWritten / 1024u
+                            val toastText = "Exported file size: $sizeWrittenInKB kB"
```
Comment (rajramsaroop) [inline DittoExportLogs/src/main/java/live/ditto/dittoexportlogs/ExportLogs.kt:75]: Put this string into the strings.xml

File: DittoExportLogs/src/main/java/live/ditto/dittoexportlogs/ExportLogs.kt [hunk 6]
```diff
@@ -1,26 +1,116 @@
...
+                            Toast.makeText(context, toastText, Toast.LENGTH_LONG).show()
+                        }
+                        catch (e: DittoError.IoError) {
+                            Log.e("DittoTools LogExporter", e.reason.toString())
+                            Toast.makeText(context, "Error exporting logs: ${e.message}", Toast.LENGTH_LONG).show()
```
Comment (rajramsaroop) [inline DittoExportLogs/src/main/java/live/ditto/dittoexportlogs/ExportLogs.kt:80]: move this to a strings.xml

File: DittoExportLogs/src/main/java/live/ditto/dittoexportlogs/ExportLogs.kt [hunk 7]
```diff
@@ -1,26 +1,97 @@
...
+
+    val now = LocalDateTime.now().format(DateTimeFormatter.ofPattern("yyyyMMdd_HHmmss"))
+    val appName = context.applicationInfo.loadLabel(context.packageManager).toString()
+
+    val fileName = "$appName-dittologs-$now.jsonl.gz"
+    val downloadsDir = Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DOWNLOADS);
+    val filePath = File(downloadsDir, fileName).absolutePath
+
+    AlertDialog(
+        modifier = modifier,
```
Comment (rajramsaroop) [inline DittoExportLogs/src/main/java/live/ditto/dittoexportlogs/ExportLogs.kt:51]: We can delete this parameter here as we never change it. And then remove the `val modifier = Modifier` on line 39 above.

File: DittoExportLogs/src/main/java/live/ditto/dittoexportlogs/ExportLogs.kt [hunk 8]
```diff
@@ -1,26 +1,97 @@
...
+        text = {
+            Text(stringResource(R.string.export_logs_message, fileName))
+        },
+
+        confirmButton = {
+            Button(
+                onClick = {
+
+
+                    coroutineScope.launch {
```
Comment (rajramsaroop) [inline DittoExportLogs/src/main/java/live/ditto/dittoexportlogs/ExportLogs.kt:65]: No need to wrap in coroutine scope as `exportLogs()` is not a suspending function (`exportLogs()` is using the `viewmodelScope` from the VM).

**Review by rajramsaroop — CHANGES_REQUESTED** [review]

File: DittoExportLogs/src/main/java/live/ditto/dittoexportlogs/ExportLogs.kt [hunk 9]
```diff
@@ -1,26 +1,97 @@
...
 import androidx.compose.ui.res.stringResource
-import live.ditto.exporter.ExportDialog
-import java.nio.file.Path
+import androidx.compose.ui.unit.dp
+import kotlinx.coroutines.launch
+import java.io.File
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
 
+@RequiresApi(Build.VERSION_CODES.O)
```
[the remark this answers is not in the document]
Comment (bplattenburg) [reply DittoExportLogs/src/main/java/live/ditto/dittoexportlogs/ExportLogs.kt:29]: @rajramsaroop Can you log a separate issue to follow up on this so it doesn't get lost?

[the remark this answers is not in the document]
Comment (rajramsaroop) [reply DittoExportLogs/src/main/java/live/ditto/dittoexportlogs/ExportLogs.kt:29]: Done: https://github.com/getditto/DittoAndroidTools/issues/141

File: DittoExportLogs/src/main/java/live/ditto/dittoexportlogs/ExportLogs.kt (hunk 8 above)
Comment (zmarkan) [reply DittoExportLogs/src/main/java/live/ditto/dittoexportlogs/ExportLogs.kt:65]: good spot - I missed when moved it out into a ViewModel

**Comment by zmarkan** [comment]
Implemented the requested changes

**Review by rajramsaroop — APPROVED** [review]

**Pull request merged**